File: sdk/typescript/src/api/system.ts
import { HTTPClient } from './http-client';
import { SystemStats, HealthStatus } from '../types';

export class SystemAPI {
  constructor(private client: HTTPClient) {}

  async getStats(): Promise<SystemStats> {
    return this.client.request<SystemStats>({
      method: 'GET',
      endpoint: '/v1/system/stats',
    });
  }

  async getHealth(): Promise<HealthStatus> {
    return this.client.request<HealthStatus>({
      method: 'GET',
      endpoint: '/v1/health',
    });
  }
}

File: sdk/rust/src/api/system.rs
use crate::client::HTTPClient;
use crate::error::TelecomError;
use crate::types::{HealthStatus, SystemStats};

pub struct SystemAPI {
    client: HTTPClient,
}

impl SystemAPI {
    pub fn new(client: HTTPClient) -> Self {
        Self { client }
    }

    pub async fn get_stats(&self) -> Result<SystemStats, TelecomError> {
        self.client.get("/v1/system/stats", None).await
    }

    pub async fn get_health(&self) -> Result<HealthStatus, TelecomError> {
        self.client.get("/v1/health", None).await
    }
}
